### PR TITLE
[MARLIN-377] Expand search functionality

### PR DESCRIPTION
**Pull request by iseessel**
From commit 79606e5 on; everything before is necessary for this ticket, but part of: https://github.com/maestrano/mnoe-admin-panel/pull/310

### CODE REVIEW

**Comment by ouranos** [comment]
@iseessel could you rebase now that #310 has been merged? Thanks

File: src/app/components/mnoe-impersonate-button/mnoe-impersonate-button.coffee [hunk 1]
```diff
@@ -43,7 +43,7 @@
         headerText: 'mnoe_admin_panel.dashboard.users.widget.local_list.modal.request_access_title'
         bodyText: 'mnoe_admin_panel.dashboard.users.widget.local_list.modal.body'
         actionCb: () -> MnoeUsers.requestAccess(vm.user).then(
-          () ->
+          (resp) ->
```
Comment (iseessel) [inline src/app/components/mnoe-impersonate-button/mnoe-impersonate-button.coffee]: not needed

File: src/app/components/mnoe-support/support-organizations-list/support-organizations-list.coffee [hunk 2]
```diff
@@ -0,0 +1,63 @@
+@App.directive('mnoeSupportOrganizationsList', ($filter, $translate, $state, MnoeOrganizations, MnoeCurrentUser, MnoeUsers, MnoeAdminConfig) ->
+  restrict: 'E',
+  scope: {
+    filterParams: '='
```
Comment (iseessel) [inline src/app/components/mnoe-support/support-organizations-list/support-organizations-list.coffee]: not needed

File: src/app/components/mnoe-support/support-users-list/support-users-list.coffee [hunk 3]
```diff
@@ -0,0 +1,54 @@
+@App.directive('mnoeSupportUsersList', ($filter, $translate, $state, MnoeOrganizations, MnoeCurrentUser, MnoeUsers, MnoeAdminConfig, toastr) ->
+  restrict: 'E',
+  scope: {
+    filterParams: '='
```
Comment (iseessel) [inline src/app/components/mnoe-support/support-users-list/support-users-list.coffee]: not needed

File: src/app/components/mnoe-support/support-users-list/support-users-list.coffee [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  templateUrl: 'app/components/mnoe-support/support-users-list/support-users-list.html',
+  link: (scope, elem) ->
```
Comment (iseessel) [inline src/app/components/mnoe-support/support-users-list/support-users-list.coffee]: elem not used.

File: src/app/components/mnoe-support/support-organizations-list/support-organizations-list.coffee [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  },
+  templateUrl: 'app/components/mnoe-support/support-organizations-list/support-organizations-list.html',
+  link: (scope, elem) ->
```
Comment (iseessel) [inline src/app/components/mnoe-support/support-organizations-list/support-organizations-list.coffee]: elem not used.

**Comment by iseessel** [comment]
@ouranos Yep, should be good -- will need to rework as per new comments.

**Comment by iseessel** [comment]
screen shot 2018-10-22 at 6 03 27 pm

File: src/app/components/mnoe-support/support-organizations-list.coffee [hunk 6]
```diff
@@ -66,22 +73,46 @@
...
+      return unless scope.organizations.externalIdSearch.length > 2
+      # Reset other search fields.
+      scope.organizations.orgNameSearch = ''
+      scope.organizations.firstNameSearch = ''
+      scope.organizations.lastNameSearch = ''
+      setSearchOrganizationsList()
+
+    scope.nameSearch = () ->
+      org = scope.organizations
+      return unless org.orgNameSearch.length >= 2 && org.firstNameSearch.length >= 2 && org.lastNameSearch.length >= 2
```
Comment (adamaziz15) [inline src/app/components/mnoe-support/support-organizations-list.coffee]: Does it need to be all three conditions? Or just one of them?

Comment (iseessel) [reply src/app/components/mnoe-support/support-organizations-list.coffee]: @adamaziz15 yep all three due to requirements.

File: src/app/services/locale-helper/locale-config.svc.coffee [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    )
+
+  setLocale = (locale) ->
+    setFallbackStack(locale)
+    $translate.use(locale)
+    amMoment.changeLocale(locale)
+
+    # api_root = "/#{locale}#{URI.api_root}"
+    # MnoeApiSvc.setBaseUrl(api_root)
+    # MnoeFullApiSvc.setBaseUrl(api_root)
```
Comment (alexmaestrano) [inline src/app/services/locale-helper/locale-config.svc.coffee]: Do we need these comments?

**Pull request merged**